Issues while building wolfSSL with certain config options

### Version

5.8.2

## wolfSSL configuration
 
I used the following configuration to add PQ support for TLS/DTLS, alongside increased debugging capabilities.
```
./configure --build=x86_64-pc-linux-gnu --host=x86_64-pc-linux-gnu --prefix=/usr/local \
  --enable-mlkem --enable-mldsa --enable-lms --enable-xmss --enable-dilithium \
  --enable-dtls --enable-dtls13 --enable-sctp --enable-intelasm --enable-intelrand \
  --enable-curve25519 --enable-ed25519 --enable-curve448 --enable-ed448 \
  --enable-opensslcoexist --enable-opensslextra --enable-hkdf --enable-debug \
  --enable-sniffer --enable-memorylog --enable-stacklog --enable-trackmemory \
  --enable-certext --enable-certreq --enable-certgen --enable-keylog-export \
  CPPFLAGS=-DWOLFSSL_SNIFFER_KEYLOGFILE
```
I took this guide as a reference for enabling keylog file export: [Building wolfSSL -- Chapter 2](https://www.wolfssl.com/documentation/manuals/wolfssl/chapter02.html#wolfssl_sniffer_keylogfile)

### Environment
- OS: Ubuntu 22.04 (**x86_64**)
- wolfSSL commit: [4280b52](https://github.com/wolfSSL/wolfssl/commit/4280b52bfffe6640f589e210742ba3c93191d261)

## SSL Sniffer Error

When building with the above configuration, the `make` process fails with the following error:

```md
make[1]: Entering directory '/home/ubuntu/wolfssl'
make[2]: Entering directory '/home/ubuntu/wolfssl'
make[2]: warning: -j13 forced in submake: resetting jobserver mode.
  CPPAS    wolfcrypt/src/src_libwolfssl_la-sha256_asm.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-sha512_asm.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-sha3_asm.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-poly1305_asm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-md5.lo
  CC       wolfcrypt/src/src_libwolfssl_la-pwdbased.lo
  CC       wolfcrypt/src/src_libwolfssl_la-pkcs12.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-aes_asm.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-aes_gcm_asm.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-chacha_asm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-chacha.lo
  CC       wolfcrypt/src/src_libwolfssl_la-chacha20_poly1305.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ecc.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_mlkem.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_mlkem_poly.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-wc_mlkem_asm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-dilithium.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-wc_mldsa_asm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_lms.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_lms_impl.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_xmss.lo
  CC       wolfcrypt/src/src_libwolfssl_la-wc_xmss_impl.lo
  CC       wolfcrypt/src/src_libwolfssl_la-curve25519.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ed25519.lo
  CC       wolfcrypt/src/src_libwolfssl_la-fe_low_mem.lo
  CPPAS    wolfcrypt/src/src_libwolfssl_la-fe_x25519_asm.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ge_low_mem.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ge_operations.lo
  CC       wolfcrypt/src/src_libwolfssl_la-curve448.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ed448.lo
  CC       wolfcrypt/src/src_libwolfssl_la-fe_448.lo
  CC       wolfcrypt/src/src_libwolfssl_la-ge_448.lo
  CC       src/libwolfssl_la-internal.lo
  CC       src/libwolfssl_la-wolfio.lo
  CC       src/libwolfssl_la-keys.lo
  CC       src/libwolfssl_la-ssl.lo
  CC       src/libwolfssl_la-tls.lo
  CC       src/libwolfssl_la-tls13.lo
  CC       src/libwolfssl_la-sniffer.lo
  CC       src/libwolfssl_la-dtls13.lo
  CC       src/libwolfssl_la-dtls.lo
  CC       wolfcrypt/test/test.o
  CC       examples/benchmark/tls_bench.o
  CC       examples/client/client-client.o
  CC       examples/echoclient/echoclient.o
  CC       examples/echoserver/echoserver.o
  CC       examples/server/server-server.o
  CC       examples/sctp/sctp-server.o
src/sniffer.c: In function ‘addSecretNode’:
src/sniffer.c:7349:33: error: array subscript is not an integer
 7349 |     node->next = secretHashTable[index];
      |                                 ^
src/sniffer.c:7350:20: error: array subscript is not an integer
 7350 |     secretHashTable[index] = node;
      |                    ^
make[2]: *** [Makefile:9007: src/libwolfssl_la-sniffer.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/ubuntu/wolfssl'
make[1]: *** [Makefile:10630: all-recursive] Error 1
make[1]: Leaving directory '/home/ubuntu/wolfssl'
make: *** [Makefile:5860: all] Error 2
```
Examining the error message, I believe there may be an issue in the sniffer's code. But let me know if I am missing something.

## Memory Test error

Additionally, there is another error I came across, when I had configured `wolfSSL`  with the `--enable-memtest` option, this error, just like the previous one, prevented the library from being built.

Here was the error message
```md
  CCLD     examples/sctp/sctp-client
wolfcrypt/src/memory.c: In function ‘xmalloc’:
wolfcrypt/src/memory.c:1553:15: error: too few arguments to function ‘malloc_function’
 1553 |         p32 = malloc_function(n + sizeof(word32) * 4);
      |               ^~~~~~~~~~~~~~~
wolfcrypt/src/memory.c: In function ‘xrealloc’:
wolfcrypt/src/memory.c:1596:15: error: too few arguments to function ‘realloc_function’
 1596 |         p32 = realloc_function(oldp32, n + sizeof(word32) * 4);
      |               ^~~~~~~~~~~~~~~~
wolfcrypt/src/memory.c: In function ‘xfree’:
wolfcrypt/src/memory.c:1642:13: error: too few arguments to function ‘free_function’
 1642 |             free_function(p32);
      |             ^~~~~~~~~~~~~
make[2]: *** [Makefile:8538: wolfcrypt/src/src_libwolfssl_la-memory.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/ubuntu/wolfssl'
make[1]: *** [Makefile:10630: all-recursive] Error 1
make[1]: Leaving directory '/home/ubuntu/wolfssl'
```

Again, this could be related to code, but correct me if I'm wrong. The error disappeared when I removed the `--enable-memtest` option from the configuration.

--- a/src/sniffer.c
+++ b/src/sniffer.c
@@ -7349,8 +7349,8 @@ static int addSecretNode(unsigned char* clientRandom,
 
     XMEMCPY(node->clientRandom, clientRandom, CLIENT_RANDOM_LENGTH);
     XMEMCPY(node->secrets[type], secret, SECRET_LENGTH);
-    node->next = secretHashTable[index];
-    secretHashTable[index] = node;
+    node->next = secretHashTable[idx];
+    secretHashTable[idx] = node;
 
 unlockReturn:
 

--- a/wolfcrypt/src/memory.c
+++ b/wolfcrypt/src/memory.c
@@ -1549,7 +1549,7 @@ void *xmalloc(size_t n, void* heap, int type, const char* func,
 #endif
 
     if (malloc_function) {
-#ifndef WOLFSSL_STATIC_MEMORY
+#if !defined(WOLFSSL_STATIC_MEMORY) && !defined(WOLFSSL_DEBUG_MEMORY)
         p32 = malloc_function(n + sizeof(word32) * 4);
 #else
         p32 = malloc_function(n + sizeof(word32) * 4, heap, type);
@@ -1592,7 +1592,7 @@ void *xrealloc(void *p, size_t n, void* heap, int type, const char* func,
     }
 
     if (realloc_function) {
-#ifndef WOLFSSL_STATIC_MEMORY
+#if !defined(WOLFSSL_STATIC_MEMORY) && !defined(WOLFSSL_DEBUG_MEMORY)
         p32 = realloc_function(oldp32, n + sizeof(word32) * 4);
 #else
         p32 = realloc_function(oldp32, n + sizeof(word32) * 4, heap, type);
@@ -1638,7 +1638,7 @@ void xfree(void *p, void* heap, int type, const char* func, const char* file,
                                                               func, file, line);
 
         if (free_function) {
-#ifndef WOLFSSL_STATIC_MEMORY
+#if !defined(WOLFSSL_STATIC_MEMORY) && !defined(WOLFSSL_DEBUG_MEMORY)
             free_function(p32);
 #else
             free_function(p32, heap, type);

